diff -Naur tar-1.29/src/tar.c tar-1.29.tpg/src/tar.c
--- tar-1.29/src/tar.c	2016-03-24 05:42:14.000000000 +0000
+++ tar-1.29.tpg/src/tar.c	2016-05-26 09:04:11.184703178 +0000
@@ -659,6 +659,8 @@
    N_("control pax keywords"), GRID+8 },
   {"label", 'V', N_("TEXT"), 0,
    N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 },
+  {"lzma", 'Y', 0, 0,
+   N_("filter the archive through lzma (depreceated flag)"), GRID+1 }
 #undef GRID
 
 #define GRID 90
@@ -1118,6 +1120,7 @@
       s = xasprintf (_("filter the archive through %s"), LZIP_PROGRAM);
       break;
 
+    case 'Y':
     case LZMA_OPTION:
       s = xasprintf (_("filter the archive through %s"), LZMA_PROGRAM);
       break;
