Index: src/tar.c
===================================================================
--- src/tar.c
+++ src/tar.c	2010-10-26 10:41:58.000000000 +0200
@@ -606,7 +606,9 @@
   {"pax-option", PAX_OPTION, N_("keyword[[:]=value][,keyword[[:]=value]]..."), 0,
    N_("control pax keywords"), GRID+8 },
   {"label", 'V', N_("TEXT"), 0,
-   N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 },
+   N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 }, 
+  {"lzma", 'Y', 0, 0,
+   N_("filter the archive through lzma (depreceated flag)"), GRID+1 },
 #undef GRID
 
 #define GRID 90
@@ -1534,6 +1536,7 @@
       set_use_compress_program_option (LZIP_PROGRAM);
       break;
 
+    case 'Y':
     case LZMA_OPTION:
       set_use_compress_program_option (LZMA_PROGRAM);
       break;
